test/cppsuite/test_harness/workload/workload_tracking.cxx

```diff
@@ -137,7 +138,11 @@ workload_tracking::do_work()
 
     free(sweep_key);
 
-    if (ret != WT_NOTFOUND)
+    /*
+     * If we get here and the test is running, we must have reached the end of the table. It is
+     * possible that the test ended while the cursor was still on a valid entry.
+     */
+    if (_running && ret != WT_NOTFOUND || (!_running && ret != WT_NOTFOUND && ret != 0))
```

Comment:
I think we need a better comment explaining this logic, I'm confused reading it myself and the comment doesn't seem to help much.

Reply:
So basically, we could end up here for two reasons, when the test is still running (`_running == true`) and when the test is done (`_running == false`).
If the test is still running, it means we have processed the whole table, we only expect `ret == WT_NOTFOUND`.
If the test is done running, it means we have interrupted the process, hence we can either have `ret == WT_NOTFOUND` (if we ended up at the end of the table at the same time the test finished) or we were on a valid entry, hence `ret == 0`.

Reply:
I hope I made it clearer in f8f3ef91a68fcc6bd77b9830d6a891979c055217, let me know!

Reply:
It doesn't really matter whether it's running or not at this point. The main concern here is that you test `ret != 0` in the second check when `!_running`. Probably we can modify this _if_ statement to something like this: `ret != 0 && ret != WT_NOTFOUND` ? The only way `ret` can be 0 is `_running == false`. In other cases, it won't hurt checking `ret != 0` too. What do you think?

Reply:
Also, if you choose to leave it as is, you'll need to add braces around the first part `(_running && ret != WT_NOTFOUND)`.

Reply:
I like your suggestion, I simplified the check in 67b424b799e1926b13857fec4a50f09cb68eaff0, thanks !